java/arcs/core/storage/StorageProxy.kt

```diff
@@ -52,36 +54,28 @@ class StorageProxy<Data : CrdtData, Op : CrdtOperationAtTime, T>(
 
     private val store = storeEndpointProvider.getStorageEndpoint()
 
+    // TODO - should we trigger sync?
     init {
         store.setCallback(ProxyCallback(::onMessage))
     }
 
-    /**
-     * Connects a [Handle]. If the handle is readable, it will receive the configured callbacks.
-     */
-    suspend fun registerHandle(handle: Handle<Data, Op, T>) {
-        // non-readers don't get callbacks, return early
-        if (!handle.canRead) return
-
-        log.debug { "Registering handle: $handle" }
-
-        val firstReader = handlesMutex.withLock {
-            readHandles.add(handle)
-            readHandles.size == 1
+    suspend fun addOnUpdate(action: (value: T) -> Unit) {
+        callbackMutex.withLock {
```

Comment:
I was wondering if we even need to have any type of runtime registration? For prodex I bet we know the handles in advance entirely at the time we setup an arc and create a storage proxy. That would really simplify the concurrency story.